File: java/Samples/Biometrics/CS/ServerSampleCS/Code/SettingsAccesor.cs
using Neurotec.Biometrics;
using Neurotec.Biometrics.Client;
using Neurotec.Samples.Properties;

namespace Neurotec.Samples.Code
{
	static class SettingsAccesor
	{
		#region Connection Settings

		public static bool UseDb
		{
			get
			{
				try { return Settings.Default.UseDB; }
				catch { return false; }
			}
			set { Settings.Default.UseDB = value; }
		}

		public static string TemplateDir
		{
			get
			{
				try { return Settings.Default.TemplateDir; }
				catch { return string.Empty; }
			}
			set { Settings.Default.TemplateDir = value; }
		}

		public static string DbServer
		{
			get
			{
				try { return Settings.Default.Server; }
				catch { return null; }
			}
			set { Settings.Default.Server = value; }
		}

		public static string DbTable
		{
			get
			{
				try { return Settings.Default.Table; }
				catch { return null; }
			}
			set { Settings.Default.Table = value; }
		}

		public static string DbUser
		{
			get
			{
				try { return Settings.Default.User; }
				catch { return null; }
			}
			set { Settings.Default.User = value; }
		}

		public static string DbPassword
		{
			get
			{
				try { return Settings.Default.Password; }
				catch { return null; }
			}
			set { Settings.Default.Password = value; }
		}

		public static string IdColumn
		{
			get
			{
				try { return Settings.Default.IDCollumn; }
				catch { return "dbid"; }
			}
			set { Settings.Default.IDCollumn = value; }
		}

		public static string TemplateColumn
		{
			get
			{
				try { return Settings.Default.TemplateCollumn; }
				catch { return "template"; }
			}
			set { Settings.Default.TemplateCollumn = value; }
		}

		public static string Server
		{
			get
			{
				try { return Settings.Default.MMAServer; }
				catch { return null; }
			}
			set { Settings.Default.MMAServer = value; }
		}

		public static int ClientPort
		{
			get
			{
				try { return Settings.Default.MMAPort; }
				catch { return 25452; }
			}
			set { Settings.Default.MMAPort = value; }
		}

		public static int AdminPort
		{
			get
			{
				try { return Settings.Default.MMAAdminPort; }
				catch { return 24932; }
			}
			set { Settings.Default.MMAAdminPort = value; }
		}

		public static string UserName
		{
			get
			{
				try { return Settings.Default.MMAUser; }
				catch { return "Admin"; }
			}
			set { Settings.Default.MMAUser = value; }
		}

		public static string Password
		{
			get
			{
				try { return Settings.Default.MMAPassword; }
				catch { return "Admin"; }
			}
			set { Settings.Default.MMAPassword = value; }
		}

		public static bool IsAccelerator
		{
			get
			{
				try { return Settings.Default.IsAccelerator; }
				catch { return false; }
			}
			set { Settings.Default.IsAccelerator = value; }
		}

		#endregion

		#region Matching parameters settings

		#region General

		public static readonly int DefaultMatchingThreshold = 48;
		public static int MatchingThreshold
		{
			get
			{
				try { return Settings.Default.MatchingThreshold; }
				catch { return DefaultMatchingThreshold; }
			}
			set { Settings.Default.MatchingThreshold = value; }
		}

		#endregion

		#region Fingers

		public static readonly NMatchingSpeed DefaultFingersMatchingSpeed = NMatchingSpeed.Low;
		public static NMatchingSpeed FingersMatchingSpeed
		{
			get
			{
				try { return Settings.Default.FingersMatchingSpeed; }
				catch { return DefaultFingersMatchingSpeed; }
			}
			set { Settings.Default.FingersMatchingSpeed = value; }
		}

		public static readonly byte DefaultFingersMaximalRotation = 128;
		public static byte FingersMaximalRotation
		{
			get
			{
				try { return Settings.Default.FingersMaximalRotation; }
				catch { return DefaultFingersMaximalRotation; }
			}
			set { Settings.Default.FingersMaximalRotation = value; }
		}

		#endregion

		#region Faces

		public static readonly NMatchingSpeed DefaultFacesMatchingSpeed = NMatchingSpeed.Low;
		public static NMatchingSpeed FacesMatchingSpeed
		{
			get
			{
				try { return Settings.Default.FacesMatchingSpeed; }
				catch { return DefaultFacesMatchingSpeed; }
			}
			set { Settings.Default.FacesMatchingSpeed = value; }
		}

		#endregion

		#region Irises

		public static readonly NMatchingSpeed DefaultIrisesMatchingSpeed = NMatchingSpeed.Low;
		public static NMatchingSpeed IrisesMatchingSpeed
		{
			get
			{
				try { return Settings.Default.IrisesMatchingSpeed; }
				catch { return NMatchingSpeed.Low; }
			}
			set { Settings.Default.IrisesMatchingSpeed = value; }
		}

		public static readonly byte DefaultIrisesMaximalRotation = 11;
		public static byte IrisesMaximalRotation
		{
			get
			{
				try { return Settings.Default.IrisesMaxRotation; }
				catch { return DefaultIrisesMaximalRotation; }
			}
			set { Settings.Default.IrisesMaxRotation = value; }
		}

		#endregion

		#region Palms

		public static readonly NMatchingSpeed DefaultPalmsMatchingSpeed = NMatchingSpeed.Low;
		public static NMatchingSpeed PalmsMatchingSpeed
		{
			get
			{
				try { return Settings.Default.PalmsMatchingSpeed; }
				catch { return DefaultPalmsMatchingSpeed; }
			}
			set { Settings.Default.PalmsMatchingSpeed = value; }
		}

		public static readonly byte DefaultPalmsMaximalRotation = 128;
		public static byte PalmsMaximalRotation
		{
			get
			{
				try { return Settings.Default.PalmsMaximalRotation; }
				catch { return DefaultPalmsMaximalRotation; }
			}
			set { Settings.Default.PalmsMaximalRotation = value; }
		}

		#endregion

		#endregion

		#region Public static methods

		public static void ResetMatchingSettings()
		{
			MatchingThreshold = DefaultMatchingThreshold;
			FingersMatchingSpeed = DefaultFingersMatchingSpeed;
			FingersMaximalRotation = DefaultFingersMaximalRotation;
			FacesMatchingSpeed = DefaultFacesMatchingSpeed;
			IrisesMatchingSpeed = DefaultIrisesMatchingSpeed;
			IrisesMaximalRotation = DefaultIrisesMaximalRotation;
			PalmsMatchingSpeed = DefaultPalmsMatchingSpeed;
			PalmsMaximalRotation = DefaultPalmsMaximalRotation;
		}

		public static void SetMatchingParameters(NBiometricClient biometricClient)
		{
			//General params
			biometricClient.MatchingThreshold = MatchingThreshold;

			//Finger params
			biometricClient.FingersMatchingSpeed = FingersMatchingSpeed;
			biometricClient.FingersMaximalRotation = FingersMaximalRotation;

			//Faces params
			biometricClient.FacesMatchingSpeed = FacesMatchingSpeed;

			//irises
			biometricClient.IrisesMatchingSpeed = IrisesMatchingSpeed;
			biometricClient.IrisesMaximalRotation = IrisesMaximalRotation;

			//palms
			biometricClient.PalmsMatchingSpeed = PalmsMatchingSpeed;
			biometricClient.PalmsMaximalRotation = PalmsMaximalRotation;
		}

		public static void SaveSettings()
		{
			Settings.Default.Save();
		}

		#endregion
	}
}

File: java/Samples/Biometrics/VB/ServerSampleVB/Code/SettingsAccesor.vb
Imports Neurotec.Biometrics
Imports Neurotec.Biometrics.Client

Friend NotInheritable Class SettingsAccesor
#Region "Connection Settings"

	Private Sub New()
	End Sub
	Public Shared Property UseDb() As Boolean
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.UseDB
			Catch
				Return False
			End Try
		End Get
		Set(ByVal value As Boolean)
			Global.Neurotec.Samples.Settings.Default.UseDB = value
		End Set
	End Property

	Public Shared Property TemplateDir() As String
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.TemplateDir
			Catch
				Return String.Empty
			End Try
		End Get
		Set(ByVal value As String)
			Global.Neurotec.Samples.Settings.Default.TemplateDir = value
		End Set
	End Property

	Public Shared Property DbServer() As String
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.Server
			Catch
				Return Nothing
			End Try
		End Get
		Set(ByVal value As String)
			Global.Neurotec.Samples.Settings.Default.Server = value
		End Set
	End Property

	Public Shared Property DbTable() As String
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.Table
			Catch
				Return Nothing
			End Try
		End Get
		Set(ByVal value As String)
			Global.Neurotec.Samples.Settings.Default.Table = value
		End Set
	End Property

	Public Shared Property DbUser() As String
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.User
			Catch
				Return Nothing
			End Try
		End Get
		Set(ByVal value As String)
			Global.Neurotec.Samples.Settings.Default.User = value
		End Set
	End Property

	Public Shared Property DbPassword() As String
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.Password
			Catch
				Return Nothing
			End Try
		End Get
		Set(ByVal value As String)
			Global.Neurotec.Samples.Settings.Default.Password = value
		End Set
	End Property

	Public Shared Property IdColumn() As String
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.IDCollumn
			Catch
				Return "dbid"
			End Try
		End Get
		Set(ByVal value As String)
			Global.Neurotec.Samples.Settings.Default.IDCollumn = value
		End Set
	End Property

	Public Shared Property TemplateColumn() As String
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.TemplateCollumn
			Catch
				Return "template"
			End Try
		End Get
		Set(ByVal value As String)
			Global.Neurotec.Samples.Settings.Default.TemplateCollumn = value
		End Set
	End Property

	Public Shared Property Server() As String
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.MMAServer
			Catch
				Return Nothing
			End Try
		End Get
		Set(ByVal value As String)
			Global.Neurotec.Samples.Settings.Default.MMAServer = value
		End Set
	End Property

	Public Shared Property ClientPort() As Integer
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.MMAPort
			Catch
				Return 25452
			End Try
		End Get
		Set(ByVal value As Integer)
			Global.Neurotec.Samples.Settings.Default.MMAPort = value
		End Set
	End Property

	Public Shared Property AdminPort() As Integer
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.MMAAdminPort
			Catch
				Return 24932
			End Try
		End Get
		Set(ByVal value As Integer)
			Global.Neurotec.Samples.Settings.Default.MMAAdminPort = value
		End Set
	End Property

	Public Shared Property UserName() As String
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.MMAUser
			Catch
				Return "Admin"
			End Try
		End Get
		Set(ByVal value As String)
			Global.Neurotec.Samples.Settings.Default.MMAUser = value
		End Set
	End Property

	Public Shared Property Password() As String
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.MMAPassword
			Catch
				Return "Admin"
			End Try
		End Get
		Set(ByVal value As String)
			Global.Neurotec.Samples.Settings.Default.MMAPassword = value
		End Set
	End Property

	Public Shared Property IsAccelerator() As Boolean
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.IsAccelerator
			Catch
				Return False
			End Try
		End Get
		Set(ByVal value As Boolean)
			Global.Neurotec.Samples.Settings.Default.IsAccelerator = value
		End Set
	End Property

#End Region

#Region "Matching parameters settings"

#Region "General"

	Public Shared ReadOnly DefaultMatchingThreshold As Integer = 48
	Public Shared Property MatchingThreshold() As Integer
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.MatchingThreshold
			Catch
				Return DefaultMatchingThreshold
			End Try
		End Get
		Set(ByVal value As Integer)
			Global.Neurotec.Samples.Settings.Default.MatchingThreshold = value
		End Set
	End Property

#End Region

#Region "Fingers"

	Public Shared ReadOnly DefaultFingersMatchingSpeed As NMatchingSpeed = NMatchingSpeed.Low
	Public Shared Property FingersMatchingSpeed() As NMatchingSpeed
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.FingersMatchingSpeed
			Catch
				Return DefaultFingersMatchingSpeed
			End Try
		End Get
		Set(ByVal value As NMatchingSpeed)
			Global.Neurotec.Samples.Settings.Default.FingersMatchingSpeed = value
		End Set
	End Property

	Public Shared ReadOnly DefaultFingersMaximalRotation As Byte = 128
	Public Shared Property FingersMaximalRotation() As Byte
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.FingersMaximalRotation
			Catch
				Return DefaultFingersMaximalRotation
			End Try
		End Get
		Set(ByVal value As Byte)
			Global.Neurotec.Samples.Settings.Default.FingersMaximalRotation = value
		End Set
	End Property

#End Region

#Region "Faces"

	Public Shared ReadOnly DefaultFacesMatchingSpeed As NMatchingSpeed = NMatchingSpeed.Low
	Public Shared Property FacesMatchingSpeed() As NMatchingSpeed
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.FacesMatchingSpeed
			Catch
				Return DefaultFacesMatchingSpeed
			End Try
		End Get
		Set(ByVal value As NMatchingSpeed)
			Global.Neurotec.Samples.Settings.Default.FacesMatchingSpeed = value
		End Set
	End Property

#End Region

#Region "Irises"

	Public Shared ReadOnly DefaultIrisesMatchingSpeed As NMatchingSpeed = NMatchingSpeed.Low
	Public Shared Property IrisesMatchingSpeed() As NMatchingSpeed
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.IrisesMatchingSpeed
			Catch
				Return NMatchingSpeed.Low
			End Try
		End Get
		Set(ByVal value As NMatchingSpeed)
			Global.Neurotec.Samples.Settings.Default.IrisesMatchingSpeed = value
		End Set
	End Property

	Public Shared ReadOnly DefaultIrisesMaximalRotation As Byte = 11
	Public Shared Property IrisesMaximalRotation() As Byte
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.IrisesMaxRotation
			Catch
				Return DefaultIrisesMaximalRotation
			End Try
		End Get
		Set(ByVal value As Byte)
			Global.Neurotec.Samples.Settings.Default.IrisesMaxRotation = value
		End Set
	End Property

#End Region

#Region "Palms"

	Public Shared ReadOnly DefaultPalmsMatchingSpeed As NMatchingSpeed = NMatchingSpeed.Low
	Public Shared Property PalmsMatchingSpeed() As NMatchingSpeed
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.PalmsMatchingSpeed
			Catch
				Return DefaultPalmsMatchingSpeed
			End Try
		End Get
		Set(ByVal value As NMatchingSpeed)
			Global.Neurotec.Samples.Settings.Default.PalmsMatchingSpeed = value
		End Set
	End Property

	Public Shared ReadOnly DefaultPalmsMaximalRotation As Byte = 128
	Public Shared Property PalmsMaximalRotation() As Byte
		Get
			Try
				Return Global.Neurotec.Samples.Settings.Default.PalmsMaximalRotation
			Catch
				Return DefaultPalmsMaximalRotation
			End Try
		End Get
		Set(ByVal value As Byte)
			Global.Neurotec.Samples.Settings.Default.PalmsMaximalRotation = value
		End Set
	End Property

#End Region

#End Region

#Region "Public static methods"

	Public Shared Sub ResetMatchingSettings()
		MatchingThreshold = DefaultMatchingThreshold
		FingersMatchingSpeed = DefaultFingersMatchingSpeed
		FingersMaximalRotation = DefaultFingersMaximalRotation
		FacesMatchingSpeed = DefaultFacesMatchingSpeed
		IrisesMatchingSpeed = DefaultIrisesMatchingSpeed
		IrisesMaximalRotation = DefaultIrisesMaximalRotation
		PalmsMatchingSpeed = DefaultPalmsMatchingSpeed
		PalmsMaximalRotation = DefaultPalmsMaximalRotation
	End Sub

	Public Shared Sub SetMatchingParameters(ByVal biometricClient As NBiometricClient)
		'General params
		biometricClient.MatchingThreshold = MatchingThreshold

		'Finger params
		biometricClient.FingersMatchingSpeed = FingersMatchingSpeed
		biometricClient.FingersMaximalRotation = FingersMaximalRotation

		'Faces params
		biometricClient.FacesMatchingSpeed = FacesMatchingSpeed

		'irises
		biometricClient.IrisesMatchingSpeed = IrisesMatchingSpeed
		biometricClient.IrisesMaximalRotation = IrisesMaximalRotation

		'palms
		biometricClient.PalmsMatchingSpeed = PalmsMatchingSpeed
		biometricClient.PalmsMaximalRotation = PalmsMaximalRotation
	End Sub

	Public Shared Sub SaveSettings()
		Global.Neurotec.Samples.Settings.Default.Save()
	End Sub

#End Region
End Class
